guide.qmd

```diff
@@ -268,8 +228,28 @@ guide_table(
 
 Life cycle management is simplified by reducing dependencies, and should comprise a central code repository.
 
+### Dependency Management
 1.  **Reduce dependencies** to simplify maintenance of your own package. Only depend on other packages that you trust and deem stable enough for the purpose, in order to avoid reinventing the wheel.
 
+
+2. **Use tools that track dependencies and their versions** as these are moving targets, i.e. they are periodically updated. Insufficient dependency management is a common reason for "it doesn't work on my computer" errors. 
```

Comment:
Hm I am not sure if we should recommend this in general. The reason being that so far I don't use this on any of my packages ... even though I am familiar with it and have trained people to use it, but only for analysis projects, and not for package development.

- Is it maybe specifically important in certain situations? And then we could add that here?
- Another reason I have doubts is that the whole repository infrastructure (based on CRAN philosophy) does not work with fixed dependency versions, but minimal dependency versions.

Reply:
Oh interesting! So how do you manage co-development of packages with many developers? It feels a bit alien to me to not track dependencies. 
For R, are you taking care of this in a less direct manner through working from the same CRAN snapshot?

Reply:
So far, co-development worked in my projects via central CI pipelines on GitHub (or GitLab), thereby having a common system of packages at the end to check the PRs. Individual developers were mostly just updating their R versions once in a while to match the central CI R versions and updating then all packages accordingly. 
Maybe we could word this as something optional to consider in case that version problems actually occur?